gdx/src/com/badlogic/gdx/graphics/g3d/decals/AlphaTestGroupStrategy.java

```diff
@@ -0,0 +1,151 @@
+        for (Array<Decal> materialGroup : materialGroups.values()) {
+            contents.addAll(materialGroup);
+        }
+
+        materialGroups.clear();
+        arrayPool.freeAll(usedArrays);
+        usedArrays.clear();
+    }
+
+    @Override
+    public void afterGroup(int group) {
+    }
+
+    @Override
+    public void beforeGroups() {
+
+        Gdx.gl.glEnable(GL20.GL_DEPTH_TEST);
+        shader.begin();
+        shader.setUniformMatrix("u_projectionViewMatrix",
+                                camera.combined);
+        shader.setUniformi("u_texture",
+                           0);
+    }
+
+    @Override
+    public void afterGroups() {
+        shader.end();
+        Gdx.gl.glDisable(GL20.GL_DEPTH_TEST);
+    }
+
+    private void createDefaultShader() {
+        String vertexShader = "attribute vec4 " + ShaderProgram.POSITION_ATTRIBUTE + ";\n" //
+                              + "attribute vec4 " + ShaderProgram.COLOR_ATTRIBUTE + ";\n" //
+                              + "attribute vec2 " + ShaderProgram.TEXCOORD_ATTRIBUTE + "0;\n" //
+                              + "uniform mat4 u_projectionViewMatrix;\n" //
+                              + "varying vec4 v_color;\n" //
+                              + "varying vec2 v_texCoords;\n" //
+                              + "\n" //
+                              + "void main()\n" //
+                              + "{\n" //
+                              + "   v_color = " + ShaderProgram.COLOR_ATTRIBUTE + ";\n" //
+                              + "   v_texCoords = " + ShaderProgram.TEXCOORD_ATTRIBUTE + "0;\n" //
+                              + "   gl_Position =  u_projectionViewMatrix * " + ShaderProgram.POSITION_ATTRIBUTE + ";"
+                              + "\n" //
+                              + "}\n";
+        String fragmentShader = "#ifdef GL_ES\n" //
+                                + "precision mediump float;\n" //
+                                + "#endif\n" //
+                                + "varying vec4 v_color;\n" //
+                                + "varying vec2 v_texCoords;\n" //
+                                + "uniform sampler2D u_texture;\n" //
+                                + "float[] bayerMatrix ={1.0f / 17.0f,  9.0f / 17.0f,  3.0f / 17.0f, 11.0f / 17.0f," //
+                                + "13.0f / 17.0f,  5.0f / 17.0f, 15.0f / 17.0f,  7.0f / 17.0f,"//
+                                + "4.0f / 17.0f, 12.0f / 17.0f,  2.0f / 17.0f, 10.0f / 17.0f, " //
+                                + "16.0f / 17.0f,  8.0f / 17.0f, 14.0f /17.0f,  6.0f / 17" + ".0f};\n" //
+                                + "\n" //
+                                + "void main()\n"//
+                                + "{\n" //
+                                + "int xPos = int(gl_FragCoord.x) % 4; \n" //
+                                + "int yPos = int(gl_FragCoord.y) % 4; \n" //
+                                + "float stipple= bayerMatrix[ xPos * (yPos*4) ]; \n" //
```

Comment:
That does look incorrect, and it also won't run on GWT (can't use variable indices into arrays). It should be documented that this is GWT-incompatible.

Reply:
Good catch! indeed, variable array index requires GLSL 130 (GLES 3.0+ / WebGL 2.0). There are possible workarounds either using a for loop to find the entry or using a texture lookup.
I was balanced to include this stipple feature from the beginning mainly because it's not really a concern of group strategy. Since implementation would bring more complexity or limitation, i'm in favor to fully remove this feature from the PR.
It could be interesting to have it but i would separate it from Decals stuff (and do it in another PR), user code could plug it in this class easily by overriding this method anyway.

Reply:
From Tettinger's post the bayer stuff is no longer going to be present anyway. I have been looking at this https://bartwronski.com/2016/10/30/dithering-part-three-real-world-2d-quantization-dithering/. I think stipple should be included though, because it is a part of the group strategy idea, that within a strategy some groups are translucent, and others opaque etc but if you remove stippling then the group strategy has no support for translucency at all when it could easily be a setting in the constructor. And i think that probably people would prefer CameraGroupStrategy or AlphaTestStrategy working entirely without needing extra stuff for the second choice, because its easy to just replace one with the other to see which is preferable.